### ISSUE TITLE
GetValueOrDefault throws exception for empty string

### ISSUE THREAD

**Issue by gpetrounrt** [dotnet/command-line-api#1571]
This might be related to https://github.com/dotnet/command-line-api/issues/1551.

With version 2.0.0-beta1.21308.1, `GetValueOrDefault<string>()` returns a valid string when passing an empty string as argument.
With version 2.0.0-beta2.21617.1, `GetValueOrDefault<string>()` throws an exception when passing an empty string as argument.

```
var argument = new Argument<string>("arg", "My arg");
argument.AddValidator(argumentResult =>
{
    var s = argumentResult.GetValueOrDefault<string>();

    return null;
});

var rootCommand = new RootCommand()
{
    argument
};
```

**Comment by jonsequitur** [comment]
Could you include the line of code that's triggering the exception? I think this has been fixed but I'd like to verify.

**Comment by gpetrounrt** [comment]
`argumentResult.GetValueOrDefault<string>();` is what triggers the exception.

**Comment by jonsequitur** [comment]
Sorry, I mean how are you invoking the parser and what are you passing on the command line? I'm not able to repro this on the latest version so it's helpful to know exactly what you're doing from end to end, especially as passing empty strings from the shell often involves some awkward escaping.

**Comment by gpetrounrt** [comment]
Here is a more complete example which you can call with `test ""` arguments:
```
public class Program
{
    public static int ProcessFile(string arg)
    {
        return 0;
    }

    public static void Main(string[] args)
    {
        var argument1 = new Argument<string>("arg1", "My arg1");
        var argument2 = new Argument<string>("arg2", "My arg2");
        argument2.AddValidator(argumentResult =>
        {
            string? s = argumentResult.GetValueOrDefault<string>();

            return null;
        });

        var rootCommand = new RootCommand()
        {
            argument1,
            argument2
        };

        rootCommand.SetHandler(
            (string arg1, string arg2) =>
            {
                return Task.FromResult(ProcessFile(arg2));
            },
            argument1,
            argument2);

        rootCommand.Invoke(args);
    }
}
```

**Comment by jonsequitur** [comment]
This is fixed by #1612.
